### DISCUSSION TITLE
Exception In Production

### DISCUSSION THREAD

**Discussion by aimonext** [filamentphp/filament#10035] [Help] [answered]
Why this exception is thrown?
![Screenshot from 2023-12-03 18-56-17](https://github.com/filamentphp/filament/assets/110348549/12a3bf7f-2b8d-42e6-bc46-6459aec8cbd2)
![Screenshot from 2023-12-03 18-56-10](https://github.com/filamentphp/filament/assets/110348549/a25a3fb7-0387-40c2-b5f9-1afa460c511f)

**Comment by aimonext** [comment]
Looks like @danharrin , the issue might be stemming from your recent commit in Filament\Resources\Pages\Concerns\InteractsWithRecord.

[Filament\Resources\Pages\ListRecords](https://github.com/filamentphp/filament/blob/86e5cce8a861f3f1d1510ed3acea35583c1e705c/packages/panels/src/Resources/Pages/ListRecords.php#L220)
[Filament\Resources\Pages\Concerns\InteractsWithRecord](https://github.com/filamentphp/filament/blob/86e5cce8a861f3f1d1510ed3acea35583c1e705c/packages/panels/src/Resources/Pages/Concerns/InteractsWithRecord.php#L112)

**Comment by danharrin** [comment, marked as answer]
The trait `InteractsWithRecord` is not used internally on `ListRecords`, so I think this is a problem in your own codebase?

I will change the signatures so they match in the next version

**Reply by aimonext** [reply]
Yes, you're correct. I've been doing it this way. Will it be changed in the future?
```php
<?php

abstract class ListCartItems extends ListRecords
{
    use InteractsWithRecord;

    protected static ?string $title = 'Cart Items';

    public function getBreadcrumb(): ?string
    {
        return __(static::$title);
    }

    public function mount($record = null): void
    {
        $this->record = $this->resolveRecord($record);

        parent::mount();
    }

    protected function getTableQuery(): Builder
    {
        return $this->getRecord()->cartItems()->getQuery();
    }
}

```

**Reply by aimonext** [reply]
Anyway, I've resolved my problem. Let me know if you have a better approach for this instead of using RelationManager. Thank you.
